From 2fdb1a1a43590122b54c7ffac3c84e3be2fe736a Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hmehrtens@maxlinear.com>
Date: Thu, 10 Jun 2021 15:12:32 +0200
Subject: [PATCH] PONRTSYS-9035: Remove the check for CONFIG_CMA_SIZE_MBYTES

We will switch to CMA memory allocated in device tree specifically for
this device, this check based on the CONFIG options will not work any
more. If we use too much CMA memory the allocation will anyway fail,
this was just a check to fail earlier.
---
 drivers/net/ethernet/lantiq/cqm/prx300/cqm.c        | 10 ----------
 drivers/net/ethernet/lantiq/cqm/prx300/cqm.h        |  1 -
 drivers/net/ethernet/lantiq/cqm/prx300/cqm_config.c |  1 -
 3 files changed, 12 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
index b993f05a9e79..5c6c64907c6e 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
@@ -3801,9 +3801,6 @@ static int init_cqm_basic(struct platform_device *pdev)
 	cbm_w32((c_base + CBM_QPUSH_BASE), 0x00260000);
 	cbm_w32((c_base + CBM_TX_CREDIT_BASE), 0x00200800);
 	size = (CQM_CPU_POOL_BUF_ALW_NUM + 4) * sizeof(u32);
-	cqm_ctrl->max_mem_alloc += size;
-	WARN_ON(cqm_ctrl->max_mem_alloc >=
-		CONFIG_CMA_SIZE_MBYTES * 1024 * 1024);
 	tmp_v = TOT_DMA_HNDL - 1;
 #ifdef CPU_POOL_ALLOWED
 	cqm_ctrl->cpu_rtn_ptr = dma_alloc_attrs(&pdev->dev, size,
@@ -3896,13 +3893,6 @@ static int pool_config(struct platform_device *pdev, int p, u32 buf_size,
 	u32 adj_size;
 
 	size = buf_size * buf_num;
-	cqm_ctrl->max_mem_alloc += size;
-	/*This is not a fool proof check, because there could be other users
-	 *allocating from CMA zone
-	 */
-	if (cqm_ctrl->max_mem_alloc >= CONFIG_CMA_SIZE_MBYTES * 1024 * 1024)
-		panic("Total mem alloc exceeds CMA zone %d\n",
-		      cqm_ctrl->max_mem_alloc);
 	cqm_ctrl->bm_buf_base[p] = dma_alloc_attrs(&pdev->dev, size,
 			&cqm_ctrl->dma_hndl_p[p],
 			GFP_KERNEL,
diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.h b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.h
index 3afbedcc1156..af0115bc598c 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.h
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.h
@@ -630,7 +630,6 @@ struct cqm_ctrl {
 	void __iomem *fsqm;
 	void __iomem *pib;
 	void __iomem *pon_dqm_cntr;
-	u32 max_mem_alloc;
 	dma_addr_t dma_hndl_qos;
 	dma_addr_t dma_hndl_p[TOT_DMA_HNDL];
 	struct dentry *debugfs;
diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm_config.c b/drivers/net/ethernet/lantiq/cqm/prx300/cqm_config.c
index 248507bfa933..3c1bb0efa717 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm_config.c
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm_config.c
@@ -219,7 +219,6 @@ struct cqm_ctrl g_prx300_ctrl = {
 	.num_intrs = 5,
 	.cqm_cfg = prx300_cqm_config,
 	.ifmux_reg_offset = 0x120,
-	.max_mem_alloc = 0,
 	.bm_buf_base[0] = NULL,
 	.bm_buf_base[1] = NULL,
 	.bm_buf_base[2] = NULL,
